src/v/cluster/rm_stm.cc

```diff
@@ -1699,19 +1700,32 @@ ss::future<> rm_stm::do_try_abort_old_tx(model::producer_identity pid) {
 
     _mem_state.expected.erase(pid);
 
+    model::tx_seq transaction_seq;
     std::optional<prepare_marker> marker;
-    auto prepare_it = _mem_state.preparing.find(pid);
-    if (prepare_it != _mem_state.preparing.end()) {
-        marker = prepare_it->second;
-    }
-    prepare_it = _log_state.prepared.find(pid);
-    if (prepare_it != _log_state.prepared.end()) {
-        marker = prepare_it->second;
+
+    if (is_transaction_ga()) {
+        auto tx_seqs_it = _log_state.tx_seqs.find(pid);
+        if (tx_seqs_it == _log_state.tx_seqs.end()) {
+            vlog(clusterlog.error, "Can not find tx_seq for pid {}", pid);
+            co_return;
+        }
+        transaction_seq = tx_seqs_it->second;
+    } else {
+        auto prepare_it = _mem_state.preparing.find(pid);
```

Comment:
we should check `_log_state.tx_seqs` and `_mem_state.tx_seqs` here too. `old'` is similar to old & new.